android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -62,55 +66,118 @@
+            this.forConsentedUsersApi = forConsentedUsersApi;
+            this.userSessionInfoProvider = userSessionInfoProvider;
+        }
+    }
+
+    public AuthenticationManager(@NonNull BridgeConfig config,
+                                 @NonNull ApiClientProvider apiClientProvider,
                                  @NonNull AccountDAO accountDAO, @NonNull ConsentDAO consentDAO) {
         checkNotNull(config);
-        checkNotNull(apiClientProvider);
         checkNotNull(accountDAO);
         checkNotNull(consentDAO);
 
         this.config = config;
         this.accountDAO = accountDAO;
         this.consentDAO = consentDAO;
-        this.apiClientProvider = apiClientProvider;
 
-        this.authenticationApi = apiClientProvider.getClient(AuthenticationApi.class);
+        this.apiClientProvider = apiClientProvider;
 
-        SignIn signIn = accountDAO.getSignIn();
-        ForConsentedUsersApi api = signIn == null ?
-                apiClientProvider.getClient(ForConsentedUsersApi.class) :
-                apiClientProvider.getClient(ForConsentedUsersApi.class, signIn);
+        this.authenticationApi = apiClientProvider.getAuthenticationApi();
 
-        this.forConsentedUsersApiAtomicReference = new AtomicReference<>(api);
+        this.authStateHolderAtomicReference = new AtomicReference<>
+                (createAuthStateFromStoredCredentials());
         listeners = Lists.newArrayList();
     }
 
+    @VisibleForTesting
+    @NonNull
+    AuthStateHolder createAuthStateFromStoredCredentials() {
+        ForConsentedUsersApi forConsentedUsersApi;
+        UserSessionInfoProvider userSessionInfoProvider = null;
+
+        ApiClientProvider.AuthenticatedClientProvider provider =
+                createAuthenticatedClientProviderFromStoredCredentials();
+        if (provider != null) {
+            forConsentedUsersApi = provider.getClient(ForConsentedUsersApi.class);
+            userSessionInfoProvider = provider.getUserSessionInfoProvider();
+        } else {
+            //unauthenticated client
+            forConsentedUsersApi = apiClientProvider.getClient(ForConsentedUsersApi.class);
+        }
+
+        return new AuthStateHolder(forConsentedUsersApi, userSessionInfoProvider);
+    }
+
+    @VisibleForTesting
+    @Nullable
+    ApiClientProvider.AuthenticatedClientProvider
+    createAuthenticatedClientProviderFromStoredCredentials() {
+        String email = accountDAO.getEmail();
+        if (!Strings.isNullOrEmpty(email)) {
+            ApiClientProvider.AuthenticatedClientProviderBuilder builder =
+                    apiClientProvider
+                            .getAuthenticatedClientProviderBuilder()
+                            .withEmail(email);
+
+            boolean hasPasswordOrSession = false;
+            String password = accountDAO.getPassword();
+            if (!Strings.isNullOrEmpty(password)) {
+                hasPasswordOrSession = true;
+                builder.withPassword(email);
```

Comment:
should this be builder.withPassword(password)?

Reply:
Yes